app/scripts/lib/nonce-tracker.js

```diff
@@ -12,15 +12,14 @@ class NonceTracker {
   // releaseLock must be called
   // releaseLock must be called after adding signed tx to pending transactions (or discarding)
   async getNonceLock (address) {
+    const pendingTransactions = this.getPendingTransactions(address)
     // await lock free
-    await this.lockMap[address]
+    if (pendingTransactions.length) await this.lockMap[address]
```

Comment:
use brackets for if-else statements
prolly should put this.lockMap[address] into a var before using it since you look it up many times

this looks suspect
```js
await this.lockMap[address]
vs
await this.lockMap[address]()
```